### PR TITLE
Added uncloaking on docking.

### PR DESCRIPTION
**Pull request by ghost**
Closes https://github.com/OpenRA/OpenRA/issues/10991.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Cloak.cs [hunk 1]
```diff
@@ -15,6 +15,7 @@
 using System.Linq;
 using OpenRA.Graphics;
 using OpenRA.Traits;
+using OpenRA.Activities;
```
Comment (obrakmann) [inline OpenRA.Mods.Common/Traits/Cloak.cs]: We sort these alphabetically, so it needs to go before `OpenRA.Graphics`.

File: OpenRA.Mods.Common/Traits/Cloak.cs [hunk 2]
```diff
@@ -178,5 +181,27 @@ void RevokeUpgrades(Actor self)
 				foreach (var u in Info.WhileCloakedUpgrades)
 					upgradeManager.RevokeUpgrade(self, u, this);
 		}
+
+		void INotifyHarvesterAction.MovingToResources(Actor self, CPos targetCell, Activity next){}
```
Comment (obrakmann) [inline OpenRA.Mods.Common/Traits/Cloak.cs]: Our coding style wants a space between the parens and the bracket, and also between the brackets, so it should look like `, Activity next) { }`

Same for the other occurrences below this line.

File: OpenRA.Mods.Common/Traits/Cloak.cs [hunk 3]
```diff
@@ -178,5 +181,27 @@ void RevokeUpgrades(Actor self)
...
+
+		void INotifyHarvesterAction.MovingToRefinery(Actor self, CPos targetCell, Activity next){}
+
+		void INotifyHarvesterAction.MovementCancelled(Actor self){}
+
+		void INotifyHarvesterAction.Harvested(Actor self, ResourceType resource){}
+
+		void INotifyHarvesterAction.Docked()
+		{
+			if (Info.UncloakOn.HasFlag (UncloakType.Dock))
```
Comment (obrakmann) [inline OpenRA.Mods.Common/Traits/Cloak.cs]: No space between `HasFlag` and the opening parens.

File: OpenRA.Mods.Common/Traits/Cloak.cs [hunk 4]
```diff
@@ -178,5 +181,27 @@ void RevokeUpgrades(Actor self)
...
+			{
+				isDocking = true;
+				Uncloak();
+			}
+		}
+
+		void INotifyHarvesterAction.Undocked()
+		{
+			isDocking = false;
```
Comment (obrakmann) [inline OpenRA.Mods.Common/Traits/Cloak.cs]: And finally here's some whitespace after the semicolon which should be removed.

**Comment by dtluna** [comment]
I apologize for that. Will use `make check` before pushing next time.

**Comment by Phrohdoh** [comment]
You'll now want to squash (via rebase) these commits into a single commit.

**Comment by Phrohdoh** [comment]
This needs to be squished again to remove f3a59de4779636d2f4f230003d6e34e806b4af1d.

**Comment by dtluna** [comment]
@Phrohdoh it is impossible: 
`Warning: the SHA-1 is missing or isn't a commit in the following line:
 - s f3a59de`

**Pull request closed without merging**